packages/protocol/test/governance/quorum.ts

```diff
@@ -0,0 +1,116 @@
+import { toFixed } from '@celo/protocol/lib/fixidity'
+import { assertEqualBN, assertLogMatches2, assertRevert } from '@celo/protocol/lib/test-utils'
+import { QuorumContract, QuorumInstance } from 'types'
+
+const Quorum: QuorumContract = artifacts.require('Quorum')
+
+// @ts-ignore
+// TODO(mcortesi): Use BN.js
+Quorum.numberFormat = 'BigNumber'
+
+contract('Quorum', () => {
+  const quorumBaseline = toFixed(50 / 100)
+  const quorumFloor = toFixed(5 / 100)
+  const updateCoefficient = toFixed(1 / 5)
+  let quorum: QuorumInstance
+  beforeEach(async () => {
+    quorum = await Quorum.new()
+    await quorum.initialize(quorumBaseline, quorumFloor, updateCoefficient)
+  })
+
+  describe('#initialize()', () => {
+    it('should have set the quorum', async () => {
+      const actualQuorum = await quorum.quorumBaseline()
+      assertEqualBN(actualQuorum, quorumBaseline)
+    })
+
+    it('should not be callable again', async () => {
+      await assertRevert(quorum.initialize(quorumBaseline, quorumFloor, updateCoefficient))
+    })
+  })
+
+  describe('#adjustedSupport()', () => {
+    const totalWeight = 100
+
+    it('should return support ratio when participation above quorum', async () => {
+      const yes = 15
+      const no = 10
+      const abstain = 30
+      const expected = toFixed(3 / 5)
```

Comment:
Could we make this `toFixed(yes / (yes + no + abstain)` so it's clear where the expected calc is coming from?